src/ImageSharp/Formats/Png/PngEncoderCore.cs

```diff
@@ -875,7 +875,7 @@ private void WriteGammaChunk(Stream stream)
             // 4-byte unsigned integer of gamma * 100,000.
             uint gammaValue = (uint)(this.gamma * 100_000F);
 
-            BinaryPrimitives.WriteUInt32BigEndian(this.chunkDataBuffer.Span.Slice(0, 4), gammaValue);
+            BinaryPrimitives.WriteUInt32BigEndian(this.chunkDataBuffer.Span[..4], gammaValue);
```

Comment:
FYI: the `Slice` here was IIRC intentional, as it produces better code than the slice. There are issues in Roslyn and runtime to care about that, but for .NET 8 these aren't resolved (when I'm not mistaken).

Reply:
We're just taking the hit here. One day Roslyn will catch up, but we need greater consistency to ease the (_massive_) maintenance burden. I'm hoping come v4/.NET8 with the SIMD migration improvements I have planned we won't notice codegen issues.